### PR TITLE
remove commit count offset and remove BRANCH_NAME

### PR DESCRIPTION
**Pull request by brthor**
cc @eerhardt

### CODE REVIEW

**Comment by brthor** [comment]
@eerhardt made the changes to use BranchInfo in update-dependencies.

File: build_projects/update-dependencies/Config.cs [hunk 1]
```diff
@@ -37,7 +40,7 @@ public class Config
         private Lazy<string> _gitHubOriginOwner;
         private Lazy<string> _gitHubUpstreamOwner = new Lazy<string>(() => GetEnvironmentVariable("GITHUB_UPSTREAM_OWNER", "dotnet"));
         private Lazy<string> _gitHubProject = new Lazy<string>(() => GetEnvironmentVariable("GITHUB_PROJECT", "core-setup"));
-        private Lazy<string> _gitHubUpstreamBranch = new Lazy<string>(() => GetEnvironmentVariable("GITHUB_UPSTREAM_BRANCH", "master"));
+        private Lazy<string> _gitHubUpstreamBranch = new Lazy<string>(() => s_branchInfo.Entries["BRANCH_NAME"]);
```
Comment (eerhardt) [inline build_projects/update-dependencies/Config.cs]: This should still use `GetEnvironmentVariable`.  The branch name from branchinfo.txt should just be the default (if no one set the variable).  It is safer, that way we can always modify these inputs without changing the code.

Comment (brthor) [inline build_projects/update-dependencies/Config.cs]: alright sounds good.

**Comment by brthor** [comment]
fixed to use the environment variable

**Comment by eerhardt** [comment]
:shipit:

**Pull request merged**